File: java/Samples/BiometricStandards/VB/ANTemplateSampleVB/RecordCreateForms/ANType8RecordCreateForm.vb
Imports System
Imports Neurotec.Biometrics.Standards
Imports Neurotec.Images
Imports Neurotec.IO

Namespace RecordCreateForms

	Partial Public Class ANType8RecordCreateForm
		Inherits ANImageBinaryRecordCreateForm
#Region "Private fields"

		Private _vectors(-1) As ANPenVector

#End Region

#Region "Public constructor"

		Public Sub New()
			InitializeComponent()

			IsLowResolution = False
			FromVectorsIr = IsrValue

			fromVectorsPanel.Enabled = rbFromVectors.Checked

			For Each value As Object In System.Enum.GetValues(GetType(ANSignatureType))
				cbSignature.Items.Add(value)
			Next value
			If cbSignature.Items.Count > 0 Then
				cbSignature.SelectedIndex = 0
			Else
				cbSignature.Enabled = False
			End If
		End Sub

#End Region

#Region "Protected methods"

		Protected Overrides Function OnCreateRecord(ByVal template As ANTemplate) As ANRecord
			Dim record As ANType8Record
			If CreateFromImage Then
				Using image As NImage = Me.Image
					record = New ANType8Record(ANTemplate.VersionCurrent, Idc, SignatureType, CType(CompressionAlgorithm, ANSignatureRepresentationType), IsrFlag, image)
				End Using
			ElseIf CreateFromVector Then
				record = New ANType8Record(ANTemplate.VersionCurrent, Idc, SignatureType, Vectors)
			ElseIf CreateFromData Then
				Using image As NImage = NImage.FromMemory(ImageData)
					record = New ANType8Record(ANTemplate.VersionCurrent, Idc, SignatureType, CType(CompressionAlgorithm, ANSignatureRepresentationType), IsrFlag, image) With {.HorzLineLength = Hll, .VertLineLength = Vll}
				End Using
			Else
				Throw New NotSupportedException()
			End If

			template.Records.Add(record)
			Return record
		End Function

		Protected Overrides Function GetCompressionFormatsType() As Type
			Return GetType(ANSignatureRepresentationType)
		End Function

#End Region

#Region "Protected properties"

		Protected ReadOnly Property CreateFromVector() As Boolean
			Get
				Return rbFromVectors.Checked
			End Get
		End Property

		Protected Property FromVectorsIr() As UInteger
			Get
				Return CUInt(Math.Round(resolutioEditBox.PpmValue))
			End Get
			Set(ByVal value As UInteger)
				resolutioEditBox.PpmValue = value
			End Set
		End Property

		Protected Property Vectors() As ANPenVector()
			Get
				Return _vectors
			End Get
			Set(ByVal value As ANPenVector())
				_vectors = value
			End Set
		End Property

#End Region

#Region "Public properties"

		Public Property SignatureType() As ANSignatureType
			Get
				Return CType(cbSignature.SelectedItem, ANSignatureType)
			End Get
			Set(ByVal value As ANSignatureType)
				cbSignature.SelectedItem = value
			End Set
		End Property

#End Region

#Region "Private form events"

		Private Sub FromVectorsRadioButtonCheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles rbFromVectors.CheckedChanged
			fromVectorsPanel.Enabled = rbFromVectors.Checked
		End Sub

		Private Sub BtnEditVectorsClick(ByVal sender As Object, ByVal e As EventArgs) Handles btnEditVectors.Click
			Using dialog As New CreateANPenVectorArrayForm()
				dialog.Vectors = Vectors
				If dialog.ShowDialog() = Windows.Forms.DialogResult.OK Then
					Vectors = dialog.Vectors
				End If
			End Using
		End Sub

#End Region
	End Class
End Namespace

File: java/Samples/BiometricStandards/CS/ANTemplateSampleCS/RecordCreateForms/ANType8RecordCreateForm.cs
using System;
using System.Windows.Forms;
using Neurotec.Biometrics.Standards;
using Neurotec.Images;
using Neurotec.IO;

namespace Neurotec.Samples.RecordCreateForms
{
	public partial class ANType8RecordCreateForm : ANImageBinaryRecordCreateForm
	{
		#region Private fields

		private ANPenVector[] _vectors = new ANPenVector[0];

		#endregion

		#region Public constructor

		public ANType8RecordCreateForm()
		{
			InitializeComponent();

			IsLowResolution = false;
			FromVectorsIr = IsrValue;

			fromVectorsPanel.Enabled = rbFromVectors.Checked;

			foreach (object value in Enum.GetValues(typeof(ANSignatureType)))
			{
				cbSignature.Items.Add(value);
			}
			if (cbSignature.Items.Count > 0)
				cbSignature.SelectedIndex = 0;
			else
				cbSignature.Enabled = false;
		}

		#endregion

		#region Protected methods

		protected override ANRecord OnCreateRecord(ANTemplate template)
		{
			ANType8Record record;
			if (CreateFromImage)
			{
				using (NImage image = Image)
				{
					record = new ANType8Record(ANTemplate.VersionCurrent, Idc, SignatureType,
						(ANSignatureRepresentationType)CompressionAlgorithm, IsrFlag, image);
				}
			}
			else if (CreateFromVector)
			{
				record = new ANType8Record(ANTemplate.VersionCurrent, Idc, SignatureType, Vectors);
			}
			else if (CreateFromData)
			{
				using (NImage image = NImage.FromMemory(ImageData))
				{
					record = new ANType8Record(ANTemplate.VersionCurrent, Idc, SignatureType,
						(ANSignatureRepresentationType)CompressionAlgorithm, IsrFlag, image)
						{
							HorzLineLength = Hll,
							VertLineLength = Vll
						};
				}
			}
			else
			{
				throw new NotSupportedException();
			}

			template.Records.Add(record);
			return record;
		}

		protected override Type GetCompressionFormatsType()
		{
			return typeof(ANSignatureRepresentationType);
		}

		#endregion

		#region Protected properties

		protected bool CreateFromVector
		{
			get
			{
				return rbFromVectors.Checked;
			}
		}

		protected uint FromVectorsIr
		{
			get
			{
				return (uint)Math.Round(resolutioEditBox.PpmValue);
			}
			set
			{
				resolutioEditBox.PpmValue = value;
			}
		}

		protected ANPenVector[] Vectors
		{
			get
			{
				return _vectors;
			}
			set
			{
				_vectors = value;
			}
		}

		#endregion

		#region Public properties

		public ANSignatureType SignatureType
		{
			get
			{
				return (ANSignatureType)cbSignature.SelectedItem;
			}
			set
			{
				cbSignature.SelectedItem = value;
			}
		}

		#endregion

		#region Private form events

		private void FromVectorsRadioButtonCheckedChanged(object sender, EventArgs e)
		{
			fromVectorsPanel.Enabled = rbFromVectors.Checked;
		}

		private void BtnEditVectorsClick(object sender, EventArgs e)
		{
			using (var dialog = new CreateANPenVectorArrayForm())
			{
				dialog.Vectors = Vectors;
				if (dialog.ShowDialog() == DialogResult.OK)
				{
					Vectors = dialog.Vectors;
				}
			}
		}

		#endregion
	}
}
